Reorganize tutorials landing page

rmlarose opened the pull request:
As per discussion with @balopat and @lamberta, moves all GitHub examples to bottom of page, emphasizing on-site tutorials.

Additionally reorganizes some tutorials to balance out the sections. Before, there was 1 beginner, 1 intermediate, and 5 advanced tutorials. Now, there is 1 beginner, 3 intermediate, and 3 advanced tutorials organized as:

- Beginner
  - Basics
- Intermediate
  - VQE
  - QAOA
  - Quantum walks
- Advanced
  - Hidden linear function
  - Rabi oscillation
  - Shor's algorithm

I think this makes more sense content-wise in addition to balancing out the sections, but let me know what you think.

95-martin-orion commented on docs/tutorials/_index.yaml:
```diff
@@ -61,9 +96,6 @@ landing_page:
         icon:
           path: /site-assets/images/cards/github-card-16x9.png
 
-    - heading: Intermediate
-      description: Use Cirq for intermediate-level subroutines and algorithms.
-      items:
```
Would it be worthwhile to preserve these headings under the Github section? I can see how it might mess with formatting, though...

rmlarose replied:
It's definitely possible to do. I'll leave it to others for a final decision.

95-martin-orion replied:
+@balopat for review.

95-martin-orion reviewed (commented):
Looks good overall.

balopat reviewed (changes requested):
1. Let's create the same beginner / intermediate / advanced in the github section too.
2. This currently fails with 

```
cirq/tutorials/_index.yaml: Exception: google3.devsite.two.compilers.yamlton.transcoders.common.InvalidYamlError: We had trouble parsing your YAML file. See valid fields at go/devsite-yaml.
ParseError: Failed to parse rows field: Message type "devsite.two.compilers.yamlton.proto.Row" has no field named "list".  Available Fields(except extensions): ['customHtml', 'background', 'buttons', 'classname', 'columns', 'description', 'foreground', 'heading', 'items', 'whitelist', 'options', 'itemsAcross', 'tld', 'experiment', 'position', 'icon']
```

rmlarose commented:
> 1. Let's create the same beginner / intermediate / advanced in the github section too.
> 2. This currently fails with
> 
> ```
> cirq/tutorials/_index.yaml: Exception: google3.devsite.two.compilers.yamlton.transcoders.common.InvalidYamlError: We had trouble parsing your YAML file. See valid fields at go/devsite-yaml.
> ParseError: Failed to parse rows field: Message type "devsite.two.compilers.yamlton.proto.Row" has no field named "list".  Available Fields(except extensions): ['customHtml', 'background', 'buttons', 'classname', 'columns', 'description', 'foreground', 'heading', 'items', 'whitelist', 'options', 'itemsAcross', 'tld', 'experiment', 'position', 'icon']
> ```

Thanks @balopat. 

1. Added these headings.
2. Fixed this by removing list field. (At least I hope - still unable to stage due to permissions, sorry about this.)

balopat commented on docs/tutorials/_index.yaml:
```diff
@@ -12,6 +12,42 @@ landing_page:
+    - heading: GitHub
+      description: See more examples on the <a href="https://github.com/quantumlib/Cirq/tree/master/examples">Cirq GitHub</a>.
+
+    - heading: Beginner
```
Missing `items`
```suggestion
    - heading: Beginner
       items:
```

balopat reviewed (approved):
I did some reorg after experimenting on the staging server - sorry that this is the situation, it is not the best :/ But I think it looks much better now!

rmlarose commented:
> I did some reorg after experimenting on the staging server - sorry that this is the situation, it is not the best :/ But I think it looks much better now!

Looks awesome!! Thanks a lot @balopat
